src/Vault.sol

```diff
@@ -1168,25 +1209,27 @@ contract Vault is ERC4626, ERC20Permit, ILiquidationSource, Ownable {
   /**
    * @notice Calculate exchange rate between the amount of assets withdrawable from the YieldVault
    *         and the amount of shares minted by this Vault.
-   * @dev We exclude the amount of yield generated by the YieldVault, so users can only withdraw their share of deposits.
-   *      Except when the vault is undercollateralized, in this case, any unclaimed yield is included in the calculation.
+   * @dev The amount of yield generated by the YieldVault is excluded from the calculation,
+   *      so users can only withdraw the amount of underlying assets they deposited.
+   *      Except when the vault is undercollateralized, in this case, any unclaimed yield is included.
    * @dev We start with an exchange rate of 1 which is equal to 1 underlying asset unit.
    * @return uint256 Exchange rate
    */
   function _currentExchangeRate() internal view returns (uint256) {
     uint256 _totalSupplyAmount = _totalSupply();
-    uint256 _totalSupplyToAssets = _convertToAssets(
+    uint256 _depositedAssets = _convertToAssets(
       _totalSupplyAmount,
       _lastRecordedExchangeRate,
       Math.Rounding.Down
     );
 
     uint256 _withdrawableAssets = _yieldVault.maxWithdraw(address(this));
 
-    // If the Vault is collateralized, we exclude the yield from the calculation
-    // so users can only withdraw their share of deposits.
-    if (_withdrawableAssets > _totalSupplyToAssets) {
-      _withdrawableAssets = _withdrawableAssets - (_withdrawableAssets - _totalSupplyToAssets);
+    // If the Vault is collateralized, yield is excluded from the withdrawable amount
+    // and users can only withdraw the amount of underlying assets they deposited.
+    // Otherwise, any unclaimed yield is included and shared proportionally amongst depositors.
```

Comment:
This comment needs to be updated I think

The additional assets are not shared proportionally, AFAIK

Reply:
It is once the Vault is undercollateralized since `_withdrawableAssets` won't be equal to `_depositedAssets`, so any underlying assets withdrawable from the YieldVault will be available and the exchange will be computed using the following formula:
`return _withdrawableAssets.mulDiv(_assetUnit, _totalSupplyAmount, Math.Rounding.Down);`

Reply:
This is a language thing; there is no yield if it's undercollateralized

Reply:
Fixed wording in this commit: https://github.com/GenerationSoftware/pt-v5-vault/pull/18/commits/81fe36a5c670947219d63287f4e205cc7d2593f6